Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3939,8 +3939,7 @@ GstElement* MediaPlayerPrivateGStreamer::createVideoSink()
 
     // Ensure the sink has the max-lateness property set.
     auto exit = makeScopeExit([this] {
-        ASSERT(m_videoSink);
```

Comment:
Why remove this but not the one before returning the method result?

Reply:
Because this patch focuses only on the holepunch case, used on Nexus Broadcom platforms. In that case, [createHolePunchVideoSink() is called and it returns null to explicitly defer to autovideosink](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L3907), so [createVideoSink() returns early](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L3973) and the final assert before returning the method result that you point out isn't called. However, [the makeScopeExit lambda](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L3940) is still called automatically before returning.

Reply:
I see... :( Maybe we can simplify this.... How about calling `createVideoSink()` only in `!USE(WPEWEBKIT_PLATFORM_BCM_NEXUS)` cases?

Reply:
That would mean splitting the holepunch setup functionality in two, so it simplifies on one place to create complexity and unneeded care of implementation details on the other.

Reply:
Sorry I don't see what you mean.

Reply:
I mean that a decision that is now taken in [createHolePunchVideoSink()](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L3907) (because it's very related to holepunch) has now to be taken "standalone" (without any holepunch context) in [createGSTPlaybin()](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L2871), without any mention to holepunch, which is the place where taking the decision to defer to autoplugging comes from.

Reply:
I see no mention of holepunch in this code, beside its current location...
```
#if USE(WPEWEBKIT_PLATFORM_BCM_NEXUS)
    // Nexus boxes use autovideosink.
    return nullptr;
#endif
```

Reply:
I give up, this spaghetti is beyond control.